service/src/main/java/bio/terra/workspace/db/ResourceDao.java

```diff
@@ -363,24 +333,59 @@ public boolean updateResource(UUID workspaceId, UUID resourceId, String name, St
       params.addValue("description", description);
     }
 
-    return DbRetryUtils.retry(() -> updateResource(workspaceId, resourceId, params));
+    return updateResourceHelper(workspaceId, resourceId, params);
   }
 
-  // -- Private Methods -- //
+  /**
+   * This is an open ended method for constructing the SQL update statement. To use it, build the
+   * parameter list making the param name equal to the column name you want to update. The method
+   * generates the column_name = :column_name list. It is an error if the params map is empty.
+   *
+   * @param params sql parameters
+   * @param workspaceId workspace identifier - not strictly necessarily, but an extra validation
+   * @param resourceId resource identifier
+   */
+  private boolean updateResourceHelper(
+      UUID workspaceId, UUID resourceId, MapSqlParameterSource params) {
+    StringBuilder sb = new StringBuilder("UPDATE resource SET ");
+
+    String[] parameterNames = params.getParameterNames();
+    if (parameterNames.length == 0) {
+      throw new InvalidDaoRequestException("Must specify some data to be updated.");
+    }
+    for (int i = 0; i < parameterNames.length; i++) {
+      String columnName = parameterNames[i];
+      if (i > 0) {
+        sb.append(", ");
```

Comment:
do you want a comma after the last entry?

Reply:
No, and I don't think this construction makes one currently